ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/RetryMechanism.java

```diff
@@ -15,11 +15,19 @@
 import io.vertx.core.buffer.Buffer;
 import io.vertx.core.http.HttpClientResponse;
 
-public interface RetryMechanism<T> {
+public abstract class RetryMechanism<T> {
 
-  class RetryException extends RuntimeException {};
+  private static final int MAX_RETRIES = 5;
```

Comment:
Keeping this as an interface might be a 'cleaner' way to go ...as it would allow easily for the option for configuring the number of retries.